File: ApiBaseCore/ApiBaseCore.Application/DTOs/UpdateUserDto.cs
using System.ComponentModel.DataAnnotations;

namespace ApiBaseCore.Application.DTOs;

public class UpdateUserDto
{
    [EmailAddress]
    public string? Email { get; set; }

    public string? Name { get; set; }

    public string? Phone { get; set; }

    public string? TaxId { get; set; }

    public string? Password { get; set; }
}

File: ApiBaseCoreJava/src/main/java/com/polancou/apibasecore/application/dtos/UpdateUserDto.java
package com.polancou.apibasecore.application.dtos;

import jakarta.validation.constraints.Email;
import lombok.Data;

@Data
public class UpdateUserDto {
    @Email
    private String email;
    
    private String name;
    
    private String phone;
    
    private String taxId;
    
    private String password;
}
